source/common/upstream/ring_hash_lb.h

```diff
@@ -48,7 +48,9 @@ class RingHashLoadBalancer : public ThreadAwareLoadBalancerBase,
 
   const RingHashLoadBalancerStats& stats() const { return stats_; }
 
-private:
+  static RingHashLoadBalancerStats generateStats(Stats::Scope& scope);
```

Comment:
Can this be moved into the protected section?

Reply:
Unfortunately `RingHashLoadBalancer::Ring` accepts a reference to `RingHashLoadBalancerStats` in its constructor. So to be able to create `DeterministicApertureLoadBalancer::Ring` from a outside context like test, we will need public access for `RingHashLoadBalancer::generateStats`